src/PathOrderMonotonic.h

```diff
@@ -0,0 +1,449 @@
+        result.push_back(polyline);
+        polyline->start_vertex = 0;
+        Point first_endpoint = polyline->converted->front();
+        Point last_endpoint = polyline->converted->back();
+        std::vector<SparsePointGridInclusiveImpl::SparsePointGridInclusiveElem<Path*>> lines_before = line_bucket_grid.getNearby(first_endpoint, COINCIDENT_POINT_DISTANCE);
+        auto close_line_before = std::find_if(lines_before.begin(), lines_before.end(), [first_endpoint, result](SparsePointGridInclusiveImpl::SparsePointGridInclusiveElem<Path*> found_path) {
+            return found_path.val->start_vertex == found_path.val->converted->size() //Don't find any line already in the string.
+                   && vSize2(found_path.point - first_endpoint) < SQUARED_COINCIDENT_POINT_DISTANCE; //And only find close lines.
+        });
+        std::vector<SparsePointGridInclusiveImpl::SparsePointGridInclusiveElem<Path*>> lines_after = line_bucket_grid.getNearby(last_endpoint, COINCIDENT_POINT_DISTANCE);
+        auto close_line_after = std::find_if(lines_after.begin(), lines_after.end(), [last_endpoint, result](SparsePointGridInclusiveImpl::SparsePointGridInclusiveElem<Path*> found_path) {
+            return found_path.val->start_vertex == found_path.val->converted->size() //Don't find any line already in the string.
+                   && vSize2(found_path.point - last_endpoint) < SQUARED_COINCIDENT_POINT_DISTANCE; //And only find close lines.
+        });
+
+        while(close_line_before != lines_before.end())
+        {
+            Path* first = close_line_before->val;
+            result.push_front(first); //Store this one in the sequence. It's a good one.
+            size_t farthest_vertex = getFarthestEndpoint(first, close_line_before->point); //Get to the opposite side.
+            first->start_vertex = farthest_vertex;
+            first->backwards = farthest_vertex != 0;
+            first_endpoint = (*first->converted)[farthest_vertex];
+            lines_before = line_bucket_grid.getNearby(first_endpoint, COINCIDENT_POINT_DISTANCE);
+            close_line_before = std::find_if(lines_before.begin(), lines_before.end(), [first_endpoint, result](SparsePointGridInclusiveImpl::SparsePointGridInclusiveElem<Path*> found_path) {
+                return found_path.val->start_vertex == found_path.val->converted->size() //Don't find any line already in the string.
+                       && vSize2(found_path.point - first_endpoint) < SQUARED_COINCIDENT_POINT_DISTANCE; //And only find close lines.
+            });
+        }
+        while(close_line_after != lines_after.end())
+        {
+            Path* last = close_line_after->val;
+            result.push_back(last);
+            size_t farthest_vertex = getFarthestEndpoint(last, close_line_after->point); //Get to the opposite side.
+            last->start_vertex = (farthest_vertex == 0) ? last->converted->size() - 1 : 0;
+            last->backwards = (farthest_vertex != 0);
```

Comment:
For consistencies sake: Either use brackets `(`, like you do here, or don't, like in the block above.

Reply:
Done secretly inside this unrelated commit: 47fe5b9e5a520de0682edad044ab79a70a281861